plugin/sycl/common/hist_util.cc

```diff
@@ -271,11 +367,48 @@ ::sycl::event BuildHistDispatchKernel(
 
   // force_atomic_use flag is used only for testing
   use_atomic = use_atomic || force_atomic_use;
+
   if (!use_atomic) {
+    const size_t th_block_size = 256;
+    const auto block_params = GetBlocksParameters(qu, size, hist_buffer->Size() / (nbins * 2));
+    const size_t block_size = block_params.block_size;
+    const size_t max_num_bins = gmat.max_num_bins;
+    using GradientPairT = xgboost::detail::GradientPairInternal<FPType>;
+
+    int eu_l1_size = 0;
+    int eu_registers_size = 0;
+    if (true) {
```

Comment:
This functionality might help here:
https://github.khronos.org/SYCL_Reference/iface/device.html#get-info

Reply:
It is possible to get info about L2 size, but L1 size should be hardcoded :(